snapcraft/meta/snap_yaml.py

```diff
@@ -139,6 +206,59 @@ def unmarshal(cls, data: Dict[str, Any]) -> "SnapMetadata":
 
         return cls(**data)
 
+    def get_provider_content_directories(self) -> List[Path]:
```

Comment:
pedantic: some debug or trace-level logs would be nice in these functions.  

I think logging is important in loops when `continue` is called to inform a developer why a section of code was skipped.

Reply:
Will add comments and debug messages as part of the next PR.